controllers/ibu_controller.go

```diff
@@ -121,16 +123,105 @@ func (r *ImageBasedUpgradeReconciler) Reconcile(ctx context.Context, req ctrl.Re
 	} else if reconcileTime == utils.StopReconciling {
 		return
 	}
+
+	if imageBasedUpgrade.Spec.Stage == ranv1alpha1.Stages.Idle {
+		return
```

Comment:
Wouldn't we potentially have work to do if the stage gets set to Idle?

Reply:
I was wondering if there is any kind of validation we could do. Feels like it's better to leave it to later stages where the spec fields are actually consumed.

Reply:
A couple thoughts...
1. I would expect any transition to "idle" to stop any in flight processes (eg stop pre-caching)
2. We should identify which transitions we "allow", at least for now, perhaps relaxing/extending later.

Reply:
1. How about deleting the CR instead? The finalizer has to do clean up either way. Putting it back to idle makes sense only if we have a good reason to reuse the CR or whatever behind it (in that case, it would be more like a pause/resume which is not really the same thing as the initial idle state
3. Definitely